### PR TITLE
AF::File#save! should persist metadata

### PR DESCRIPTION
**Pull request by jcoyne**
Previously metadata was only persisted when `save()` was called,
`save!()` was not persisting.

### CODE REVIEW

File: lib/active_fedora/file.rb [hunk 1]
```diff
@@ -176,6 +170,12 @@ def prefix(path)
 
     private
 
+      def create_or_update(*options)
+        super.tap do
+          metadata.save if metadata.changed?
```
Comment (cbeer) [inline lib/active_fedora/file.rb]: (Without looking), should this be `#save!` and/or do we need some error handling logic if `#save` fails?

Comment (jcoyne) [inline lib/active_fedora/file.rb]: It's possible, but that should be in a different PR.  This PR is about moving the logic to be called when `save` or `save!` are called.

**Comment by awead** [comment]
👍 if you can link a new issue for the `save!` question.

File: lib/active_fedora/file.rb (hunk 1 above)
Comment (jcoyne) [inline lib/active_fedora/file.rb]: `ActiveFedora::WithMetadata::MetadataNode` does not respond to `save!` and does not have any error handling.

**Comment by jcoyne** [comment]
@awead I don't really even know where to begin on this ticket: https://github.com/projecthydra/active_fedora/issues/1092

**Comment by awead** [comment]
@jcoyne yeah, it's possible we might just close it, but the ticket would have the discussion. I'll merge this PR now since we've marked the save! issue for later resolution, even if the resolution is we don't want to do anything.

File: lib/active_fedora/file.rb (hunk 1 above)
Comment (cbeer) [inline lib/active_fedora/file.rb]: 👌
